src/riak_kv_index_hashtree.erl

```diff
@@ -370,7 +403,8 @@ handle_cast(build_failed, State) ->
     {noreply, State2};
 handle_cast(build_finished, State) ->
     State2 = do_build_finished(State),
-    {noreply, State2};
+    State3 = do_inform_waiting(State2),
```

Comment:
Also, we should probably remove the wait for lock Pid after notifying that process that the build failed.